tests/system/Test/TestCaseEmissionsTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace CodeIgniter\Test;
+
+use CodeIgniter\HTTP\Response;
+use CodeIgniter\HTTP\Exceptions\HTTPException;
+use Config\App;
+
+/**
+ * This test suite has been created separately from
+ * TestCaseTest because it messes with output
+ * buffering from PHPUnit, and the individual
+ * test cases need to be run as separate processes.
+ */
+class TestCaseEmissionsTest extends \CIUnitTestCase
+{
+	//--------------------------------------------------------------------
+
+	/**
+	 * This needs to be run as a separate process, since phpunit
+	 * has already captured the "normal" output, and we will get
+	 * a "Cannot modify headers" message if we try to change
+	 * headers or cookies now.
+	 * 
+	 * Furthermore, this test needs to flush the output buffering
+	 * that might be in progress, and start our own output buffer
+	 * capture.
+	 * 
+	 * This test includes a basic sanity check, to make sure that
+	 * the body we thought would be sent actually was.
+	 * 
+	 * @runInSeparateProcess
+	 */
+	public function testHeaderEmitted()
+	{
+		$response = new Response(new App());
+		$response->pretend(FALSE);
+
+		$body = 'Hello';
+		$expected = $body;
+
+		// what do we think we're about to send?
+		$response->setCookie('foo', 'bar');
+		$this->assertTrue($response->hasCookie('foo'));
+		$this->assertTrue($response->hasCookie('foo', 'bar'));
+
+		// send it
+		$response->setBody($body);
+
+		ob_start();
```

Comment:
I think this needs to have the `PHP_OUTPUT_HANDLER_CLEANABLE` flag set? According to the docs: 

> The output buffer must be started by ob_start() with PHP_OUTPUT_HANDLER_CLEANABLE flag. Otherwise ob_get_clean() will not work.